docs_src/design/ucr/0001-System-Events-for-Devices.md

```diff
@@ -0,0 +1,49 @@
+
+### Motivation
+When a new camera device is added to the system only Core Metadata and the Device Service managing the new camera device are aware. There are use cases when other parts of the  system need to know when a new device has been added to the system. This UCR will focus on the camera management uses case which illustrates the need for these new System Events.
+
+### Target Users
+- Device User
+- Software Deplorer
+- Software Integrator
+- Deployment Customer
+
+### Description
+System Events (aka Control Plane Events - CPE) are events generated by the system when there are changes in part of the system that are important for other parts of the system to know about. This UCR will focus on the Device System Events use case as related to camera management. These Device System Events could be utilized by many other use cases in similar manner.
+
+The new EdgeX USB and ONVIF camera Device Services (not yet released) implement auto provisioning which detects when a new camera device has been connected to USB or added to the network. New Device objects are created in Core Metadata for the new camera devices that have been auto provisioned. The auto provisioning also detects existing known camera devices, determines if there have been changes to the device details, such as IP address, and updates the Device object in Core Metadata with any changes. Device objects can also be manually deleted from Core Metadata once camera devices have been permanently disconnected.
+
+A camera management application service needs to know when a new camera has been added so that it can initiate AI/ML processing on the stream from the new camera. The service also needs to know when an existing camera device has been updated so that it can make any needed adjustments such as restart the AI/ML processing using the new IP address of the camera. Finally the service needs to know when an existing camera device has been removed so that it can stop the AI/ML processing for the removed camera.
+
+### Existing solutions
+Parts of the system (i.e. application service) must poll Core Metadata for list of devices to determine if a device has been added, update or deleted. To do this it must keep its own list or Device objects for the know devices to make these determinations.
+
+As a temporary stop gap for the initial upcoming release of the new ONVIF camera device service, an enhancement was added which publishes an EdgeX Event/Reading when a new camera device has been added, updated or modified. The Reading contains the information about the event type and the device name. This is improper use of the EdgeX Event/Reading which is intended for readings from devices, not System Events. This feature in the ONVIF Camera Device Service will be removed once System Events for Devices are in place.
+
+### Requirements
+- Added/Updated/Deleted Device System Events are created and sent in a manner that other parts of the system can subscribe and receive them.
```

Comment:
Is this just about being able to subscribe to all device system events or subscribe by event type and device name too?

Reply:
For this UC it would be Device Events for devices managed by the ONVIF or USB camera services. I'll add details on subscribe requirements.